keyring_linux.go

```diff
@@ -2,15 +2,24 @@ package keyring
 
 import (
 	"fmt"
+
 	"github.com/godbus/dbus"
-	"github.com/zalando/go-keyring/secret_service"
+	kw "github.com/zalando/go-keyring/kwallet"
+	ss "github.com/zalando/go-keyring/secret_service"
 )
 
 type secretServiceProvider struct{}
```

Comment:
I would rather add a member `typ`, which defaults to the current one and can be set to "kwallet" or something  like that.
Then you can check in methods of secretServiceProvider, without changing the default as you do right now.

A step further would be to create an interface and separate both types: secretServiceProvider and kwalletProvider.
I think this is the preffered way.

Reply:
just to understand you correctly, the detection of the actual backend should not be done each time a Get/Set/Delete method is called? but still automatically?

Reply:
Yes not every time in Get/Set/...